src/redis_hash.cc

```diff
@@ -96,20 +100,24 @@ rocksdb::Status Hash::IncrByFloat(const Slice &user_key, const Slice &field, flo
   InternalKey(ns_key, field, metadata.version).Encode(&sub_key);
   if (s.ok()) {
     std::string value_bytes;
+    std::size_t idx = 0;
     s = db_->Get(rocksdb::ReadOptions(), sub_key, &value_bytes);
     if (!s.ok() && !s.IsNotFound()) return s;
     if (s.ok()) {
       try {
-        old_value = std::stof(value_bytes);
+        old_value = std::stof(value_bytes, &idx);
       } catch (std::exception &e) {
         return rocksdb::Status::InvalidArgument(e.what());
       }
+      if (isspace(value_bytes[0]) || idx != value_bytes.size()) {
+        return rocksdb::Status::InvalidArgument("value is not an float");
+      }
       exists = true;
     }
   }
-  if ((increment < 0 && old_value < 0 && increment < (std::numeric_limits<float>::lowest()-old_value))
-      || (increment > 0 && old_value > 0 && increment > (std::numeric_limits<float>::max()-old_value))) {
-    return rocksdb::Status::InvalidArgument("increment or decrement would overflow");
+  if (increment+old_value < std::numeric_limits<float>::lowest()
+      || increment+old_value > std::numeric_limits<float>::max()) {
```

Comment:
we should use `increment > (std::numeric_limits<float>::max()-old_value)` to check whether it's overflow or not, since the `increment+old_value` may overflow and it would smaller than std::numeric_limits<float>::max().

Reply:
Thanks @git-hulk 
I think the overflow behavior of float/double is different with integer.
`std::numeric_limits<float>::max() -/+ 1` still equal `std::numeric_limits<float>::max() `
`std::numeric_limits<float>::min() -/+ 1` still equal `std::numeric_limits<float>::min() `
It seems the overflowed value of float/double is Inf, so we can only check the result is Inf or Nan
https://github.com/redis/redis/blob/unstable/src/t_string.c#L643